test/docs/AD_backend.md

```diff
@@ -1,34 +1,85 @@
+```
+
+Takeaways:
+- the `:optimized` backend (with reverse mode for Hessian) is much better than full forward mode.
+- manual sparse pattern seems to give even better performance for larger problems. This is likely due to the increasing cost of computing the Hessian sparsity in terms of allocations and time. This observation is consistent with the comparison with Jump that seems to use a different, less sparse but faster method for the Hessian.
+
+Standard benchmark:
+| Trapeze | default | optimized | manual  |
+|---------|---------|-----------|---------|
+| 250     | 49.7    | 0.9       | 1.5     |
+| 500     |         | 2.4       | 3.5     |
+| 1000    |         | 5.6       | 6.4     |
+| 2500    |         | 23.9      | 23.9    |
+| 5000    |         | 89.6      | 56.3    |
+| 7500    |         | 225.4     | 85.9    |
+| 10000   |         | 526.3     | 102.4   |
+
+Sparsity details: goddard_all Trapeze (1000 and 10000 steps)
+| transcription | optimized | manual  | optimized | manual |
+|---------------|-----------|---------|-----------|--------|
+| NLP vars      | 4005      | 4005    | 40005     | 40005  |
+| NLP cons      | 6007      | 6007    | 60007     | 60007  |
+| Hess nnz      | 11011     | 30024   | 110011    | 300024 |
+| H sparsity    | 99.86%    | 99.63%  | 99.99%    | 99.96% |
+| Jac nnz       | 28011     | 42043   | 280011    | 420043 |
+| J sparsity    | 99.88%    | 99.83%  | 99.99%    | 99.98% |
+| allocs        | 1.16GB    | 106MB   | 71.56GB   | 4.55GB |
+| time          | 750ms     | 85ms    | 64.7s**   | 3.8s   |
+
+** hessian accounts for 59 out of total 65s
+```
+julia> direct_transcription(goddard_all().ocp, grid_size=10000, show_time=true);
+gradient backend ADNLPModels.ReverseDiffADGradient: 0.000137972 seconds;
+hprod    backend ADNLPModels.ReverseDiffADHvprod: 0.314931491 seconds;
+jprod    backend ADNLPModels.ForwardDiffADJprod: 2.2412e-5 seconds;
+jtprod   backend ADNLPModels.ReverseDiffADJtprod: 0.612174104 seconds;
+jacobian backend ADNLPModels.SparseADJacobian: 0.425535048 seconds;
+hessian  backend ADNLPModels.SparseReverseADHessian: 58.450146911 seconds;
+ghjvprod backend ADNLPModels.ForwardDiffADGHjvprod: 4.339e-6 seconds.
+```
+
+| solve         | optimized | manual  | optimized | manual |
+|---------------|-----------|---------|-----------|--------|
+| iterations    | 42        | 28      | 51        | 29     |
+| allocs        | 2.0GB     | 1.2GB   | 87.5GB    | 16.9GB |
+| time          | 2.5s      | 2.5s    | 151.0s*** | 42.4s  |
 
-| Trapeze | default | optimized |
-|---------|---------|-----------|
-| 250     | 43.3    | 1.0       |
-| 500     | 176.2   | 2.4       |
-| 1000    | 926.0   | 7.1       |
-| 2500    |         | 31.8      |
-| 5000    |         |           |
+*** building the hessian is one third of the total solve time...
```

Comment:
I will pass `show_time` to the constructor of the sparse Jacobians and Hessians such that you can know how much time is spend in the sparsity detection (https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/issues/325).